fairlearn/metrics/_metric_frame.py

```diff
@@ -382,19 +382,46 @@ def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
-        if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                max_val = self._by_group[m].max()
-                result[m] = max_val
-        else:
-            result = self._by_group.groupby(level=self.control_levels).max()
+        if errors not in ("raise", "coerce"):
+            raise ValueError("invalid error value specified")
```

Comment:
Can you list out the valid values in the message? Also, can you make these strings constants at the top of the file

Reply:
Yes will do